pkg/apiserver/rest/apis/v1/types.go

```diff
@@ -20,6 +20,58 @@ import (
 	"time"
 )
 
+// CreateAddonRequest defines the format for addon create request
+type CreateAddonRequest struct {
+	Name string `json:"name" validate:"required"`
+
+	Description string `json:"description,omitempty"`
+
+	Icon string `json:"icon"`
+
+	Initializer string `json:"initializer"`
+}
+
+// ListAddonResponse defines the format for addon list response
+type ListAddonResponse struct {
+	Addons []AddonMeta `json:"addons"`
+}
+
+// AddonStatusResponse defines the format of addon status response
+type AddonStatusResponse struct {
+}
+
+// AddonPhase defines the phase of an addon
+type AddonPhase string
+
+const (
+	// AddonPhaseDisabled indicates the addon is disabled
```

Comment:
What does disable mean here? Uninstall or logical disable?

Reply:
uninstall. Following the CLI